modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2558,6 +2559,11 @@ def concat(
             left_parts = self._partitions
             right_parts = [o._partitions for o in others]
             new_lengths = self._row_lengths_cache
+            other_widths = [width for o in others for width in o._column_widths_cache]
```

Comment:
I think we can also compute `new_widths` in the `else` case below, because `_copartition` shouldn't change sizes along `axis == 1`.